modules/graphstudio/pages/map-data-to-graph.adoc

```diff
@@ -131,16 +128,42 @@ Data files, after decompression, must be in either csv or parquet format.
 [NOTE]
 ====
 TigerGraph supports loading from archived and compressed S3 files directly.
-Currently supported file extensions includes zip, tar.gz, tgz and tar.
+Currently supported file extensions include zip, tar.gz, tgz and tar.
 GraphStudio detects the file extension and automatically chooses the corresponding file format.
 If the file is encoded with one of these formats but has a non-standard file extension, you can manually specify the file format.
 ====
 
-After clicking the btn:[ADD] button, an S3 file icon will appear on the working panel:
+After clicking the btn:[ADD] button, your data will appear in the working panel.
+
+Continue with <<Map data files to vertex type or edge type>>
+
+=== Create a GCS data source
+
+After clicking btn:[Local File] when selecting a data source type, you will be prompted to enter a custom name for your GCS data source.
+
+image::add-data-source-from-gcs.png[]
+
+Underneath the name line, upload your GCS account key file. Google provides a guide on generating and downloading key files at this link: link:https://cloud.google.com/iam/docs/creating-managing-service-account-keys#getting_a_service_account_key[Getting a service account key].
+
+After you enter your key, enter the gsutil URI for your data file in your Google Cloud Storage bucket.
+
+image::gcs-data-source.png[]
+
+In addition to single files, TigerGraph also supports loading an entire folder by entering the gsutil URI for that folder.
+
+All data files in this folder must share the same data schema. When uploading a folder, the file preview is limited to the first 10 lines.
```

Comment:
```suggestion
All data files in this folder must share the same data schema. The folder preview is limited to the first 10 lines when adding a folder.
```

In fact, we always limit the preview to the first 10 lines, either for adding a file or a folder. The tricky thing is that the user might expect to preview all the files in the folder, but we only support previewing the first 10 lines in this folder.

- If the folder has only 1 file, which has > 10 lines, we will show the first 10 lines.
- If the folder has 3 files, each has > 10 lines, we will show the first 10 lines of the first file.
- If the folder has 3 files, each has 3 lines, we will show all 9 lines.

I'd imagine that the users would have a lot of confusion around this topic. It'll be great if we can explain it in detail in the user doc.